### PR TITLE
Add a mypy self test

### PR DESCRIPTION
**Pull request by srittau**
Rename mypy_selftest to mypy_test_suite

Cf. #4333

### CODE REVIEW

File: tests/mypy_test_suite.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        dirpath = Path(tempdir)
+        subprocess.run(['python2.7', '-m', 'pip', 'install', '--user', 'typing'], check=True)
+        subprocess.run(['git', 'clone', '--depth', '1', 'git://github.com/python/mypy',
+                        str(dirpath / 'mypy')], check=True)
+        subprocess.run([sys.executable, '-m', 'pip', 'install', '-U', '-r',
+                        str(dirpath / 'mypy/test-requirements.txt')], check=True)
+        shutil.copytree('stdlib', str(dirpath / 'mypy/mypy/typeshed/stdlib'))
+        shutil.copytree('third_party', str(dirpath / 'mypy/mypy/typeshed/third_party'))
+        try:
+            subprocess.run(['pytest', '-n12'], cwd=str(dirpath / 'mypy'), check=True)
```
Comment (hauntsaninja) [inline tests/mypy_test_suite.py:22]: maybe this should be `-n auto`?

File: tests/mypy_selftest.py [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
-        shutil.copytree('stdlib', str(dirpath / 'mypy/mypy/typeshed/stdlib'))
-        shutil.copytree('third_party', str(dirpath / 'mypy/mypy/typeshed/third_party'))
+        subprocess.run(
+            ["git", "clone", "--depth", "1", "--branch", f"v{MYPY_VERSION}", "git://github.com/python/mypy", str(dirpath)],
+            check=True,
+        )
         try:
-            subprocess.run(['pytest', '-n12'], cwd=str(dirpath / 'mypy'), check=True)
+            subprocess.run([sys.executable, "-m", "pip", "install", f"mypy=={MYPY_VERSION}"], check=True)
+            subprocess.run(["mypy", "--custom-typeshed-dir", ".", dirpath / "mypy", dirpath / "mypyc"], check=True)
```
Comment (hauntsaninja) [inline tests/mypy_selftest.py:21]: This doesn't pass for me against mypy master. The command mypy uses is `mypy --config-file mypy_self_check.ini -p mypy -p mypyc`

File: tests/mypy_test_suite.py (hunk 1 above)
Comment (srittau) [reply tests/mypy_test_suite.py:22]: I just moved this file from the old `mypy_selftest.py`. I'd consider changes to it out of scope of this PR.

File: tests/mypy_selftest.py (hunk 2 above)
Comment (srittau) [reply tests/mypy_selftest.py:21]: Let's try this config file.

**Comment by srittau** [comment]
Still failing. We'd need to investigate further if we decide that this is a viable way to go forward.

**Pull request merged**